Zend/zend_inheritance.c

```diff
@@ -673,6 +706,40 @@ static void do_inherit_property(zend_property_info *parent_info, zend_string *ke
 				child_info->offset = parent_info->offset;
 			}
 		}
+
+		if (!(child_info->flags & ZEND_ACC_CHANGED)) {
```

Comment:
My understanding of `CHANGED` is a bit sketchy, but I don't think this check is correct. Consider:

```php
<?php
class Foo {
    private string $a;
}
class Bar extends Foo {
    public int $a;
}
class Baz extends Bar {
    public bool $a;
}
```

Here Baz::$a should trigger an error, but (according to 3v4l) it doesn't.